WNPRC_EHR/src/weight/components/AnimalInfoPane.tsx

```diff
@@ -0,0 +1,135 @@
+import * as React from "react";
+import { Table } from "react-bootstrap";
+import Spinner from "../../components/Spinner";
+import {PaneProps} from "../typings/main";
+
+
+/**
+ * Displays animal info given a successful state.
+ */
+const AnimalInfoPane: React.FunctionComponent<PaneProps> = props => {
+  const { animalInfo, infoState } = props;
+
+  if (infoState == "waiting") {
+    return <div id="animal-info-empty">Select a record.</div>;
+  }
+
+  if (infoState == "loading-unsuccess") {
+    return <div id="animal-info-empty">Animal not found.</div>;
+  }
+
+  if (infoState == "loading") {
+    return (
+      <div id="animal-info-empty">
+        <Spinner text={"Loading..."} />
+      </div>
+    );
+  }
+
+  if (infoState == "loading-success") {
```

Comment:
I though we move away from hard coding html in the type script files?

Reply:
Not sure what you mean here, this basically determines what html should render across different "animal info" states, e.g., animal not found, animal found, etc.

Reply:
Ah I think I know what you mean, referring to the part below. Yes, we did do this change for the animal history version but not for this, I can add this as a TODO.

Reply:
Yeah that will be good, I remember you mentioned that we went away from hard coding and using a view to custom the animal abstract